lm-agent/lm_agent/workload_managers/slurm/cmd_utils.py

```diff
@@ -145,6 +145,30 @@ async def scontrol_show_lic():
     return output
 
 
+async def get_lead_host(nodelist):
+    """
+    Get the job's lead host from the nodelist.
+
+    The lead host is the first node in the nodelist.
+    The nodelist can contain multiple lists of nodes inside square brackets.
+    """
+    cmd = [
+        SCONTROL_PATH,
+        "show",
+        "hostnames",
+        nodelist,
+    ]
+
+    proc = await asyncio.create_subprocess_shell(
+        shlex.join(cmd), stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.STDOUT
+    )
+
+    stdout, _ = await asyncio.wait_for(proc.communicate(), CMD_TIMEOUT)
+    output = str(stdout, encoding=ENCODING)
+
+    return output.split("\n")[0]
```

Comment:
Optional comment: It may be nice to raise some error if it results in an empty string

Reply:
Good catch, I used `require_condition` from `buzz` to raise an exception if the `lead_host` is empty and also added a test for this part.